metricbeat/module/prometheus/collector/collector.go

```diff
@@ -63,12 +65,12 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	}, nil
 }
 
-func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
+// Fetch fetches data and reports it
+func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
 	families, err := m.prometheus.GetFamilies()
 
 	if err != nil {
-		reporter.Error(fmt.Errorf("Unable to decode response from prometheus endpoint"))
-		return
+		return fmt.Errorf("Unable to decode response from prometheus endpoint")
```

Comment:
Nit. Error message should be lowercase
```suggestion
		return fmt.Errorf("unable to decode response from prometheus endpoint")
```